src/Mod/TechDraw/Gui/QGIViewBalloon.cpp

```diff
@@ -493,29 +492,51 @@ void QGIViewBalloon::balloonLabelDragged(bool ctrl)
     DrawView* balloonParent = getSourceView();
     if (balloonParent != nullptr) {
         scale = balloonParent->getScale();
+        double xPos = Rez::appX(balloonLabel->X() / scale);
+        double yPos = Rez::appX(balloonLabel->Y() / scale);
+        // redraw the balloon at the new position
+        // note that we don't store the new position to the X/Y properties
+        // since the dragging is not yet finished
+        draw(true);
+    }
+    // store if origin is also moving to be able to later calc new origin and update feature
+    if (ctrl) {
+        m_originDragged = true;
     }
+}
+
+void QGIViewBalloon::balloonLabelDragFinished()
+{
+    // stores the final drag position for undo
+
+    auto dvb(dynamic_cast<TechDraw::DrawViewBalloon*>(getViewObject()));
+    if (dvb == nullptr)
```

Comment:
Stylistic only: Core Guidelines suggest using `if (dvb)` rather than `if (dvb == nullptr)`. Obviously not critical, but if you end up needing to make any other changes, may as well. (See http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es87-dont-add-redundant--or--to-conditions)

Reply:
Oh, in the past when I just try to fix a bug I took the existing code and changed often its style because the nice Visual Studio IDE informed me about style issues. However, I got feedback that this increased the diff unnecessarily.
Meanwhile I am on your side, also fixes should be used to improve the coding style, so I will change the style now for this PR.

Reply:
I agree that changes to unrelated lines clutter the diffs, but IMO when you are already changing a particular line, may as well make all of the changes :). Opinions differ, of course.

Reply:
I know I've given you feedback before but I think that was when there where changes in otherwise untouched functions. In this case I agree with @chennes as you already touches the line 👍 

If there are a lot of unrelated style changes I want to do in a function I usually try to split it into a separate commit to make it easier on the reviewer, but I guess opinion differ here too :)